packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils-admin

```diff
@@ -62,49 +62,66 @@ local function firmware_upgrade(msg)
         return utils.printJson({status = 'error',
                                 message = "Firmware file not found"})
     end
+    local metadata = msg.metadata or {}
+    local backup = ""
+    if not msg.preserve_config then
+        backup = "DO_NOT_BACKUP=1"
```

Comment:
isn't this an implicit message passing approach?
wouldn't it better to be added as a flag in lime-sysupgrade?

Reply:
It is not implicit in my opinion, but certainly it is not nice to have to do it using env variables. The thing is that lime-sysupgrade is a bash script that has to be reworked a lot to be able to receive parameters.

Reply:
ok... maybe for future work then... thanks!